src/main/scala/io/iohk/ethereum/utils/ByteUtils.scala

```diff
@@ -70,21 +76,93 @@ object ByteUtils {
   }
 
 
-  def bytesToInts(bytes: Array[Byte]): Array[Int] = {
-    val buffer =  ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN)
-    val ints = new Array[Int](bytes.length / 4)
-    buffer.asIntBuffer().get(ints)
-    ints
+  def bytesToInts(bytes: Array[Byte], bigEndian: Boolean): Array[Int] = {
+    val ret = new Array[Int](bytes.length / 4)
+    bytesToIntsMut(bytes, ret, bigEndian)
+    ret
   }
 
-  def intsToBytes(ints: Array[Int]): Array[Byte] = {
-    val buf = ByteBuffer.allocate(ints.length * 4).order(ByteOrder.LITTLE_ENDIAN)
-    buf.asIntBuffer().put(ints)
-    buf.array()
+  def intsToBytes(ints: Array[Int], bigEndian: Boolean): Array[Byte] = {
+    val ret = new Array[Byte](ints.length * 4)
+    intsToBytesMut(ints, ret, bigEndian)
+    ret
   }
 
   def getIntFromWord(arr: Array[Byte]): Int = {
     ByteBuffer.wrap(arr, 0, 4).order(ByteOrder.LITTLE_ENDIAN).getInt
   }
 
+  def intsToBytesMut(arr: Array[Int], b: Array[Byte], bigEndian: Boolean) {
```

Comment:
I would suggest to add comments to all this methods with `var`s and `while`s  or one on top of the file - that it is because of performance, so it wouldn't be changed if someone would think of refactoring